### ISSUE TITLE
Bug with internal links containing '&' and '|' characters not linking correctly

### ISSUE THREAD

**Issue by londonsangongzi** [verocloud/obsidian-mindmap-nextgen#133]
Hi all, I encountered a bug while using internal links, such as 

> `[[A & B]]`, `[[A & B|AB]]`, `[AB](<A & B.md>)`

which cannot link to related .md files properly due to the special characters "&" and "|".

Environment: 
obsidian 1.1.9, mindmap-nextgen 1.6.0

Fix suggest:
I've identified the issue within the `replaceInternalLinks` function, and suggest the following changes:
1. `const linkText` --> `var linkText`
2. `const linkPath` --> `var linkPath`
3. Comment out the following line of code: 
``const url = `obsidian://open?vault=${vaultName}&file=${isWikiLink ? encodeURI(obsidian.getLinkpath(linkPath)) : linkPath}`;``
4. Add the following code in its place:
```
var url = "";
if(isWikiLink){
  //"&": [[A & B]]
  linkPath = linkPath.replace(/&amp;/g, "&");
  //"|": [[A & B|AB]]
  const regex = /^(.*?)\|(.*)$/;
  const match_link_text = regex.exec(linkPath);
  if(match_link_text && match_link_text.length === 3){
	linkText = match_link_text[2];
	linkPath = match_link_text[1].trim();
  }
  url = `obsidian://open?vault=${vaultName}&file=${encodeURIComponent(obsidian.getLinkpath(linkPath))}`;
}
else{
  //"&": [AB](<A & B.md>)
  linkPath = linkPath.replace(/&amp;/g, "%26");
  url = `obsidian://open?vault=${vaultName}&file=${linkPath}`;
}
```
As a new user on GitHub, I'm still learning how to report bugs and contribute to the community. Any help or guidance would be greatly appreciated. Thank you!

Best regards,
Jeff

**Comment by james-tindal** [comment]
Thanks Jeff. Since you've figured out a solution yourself, you could make a pull request

Click the Fork button in the upper right. <img width="144" alt="image" src="https://user-images.githubusercontent.com/10291002/219661455-99996e08-ca31-4f1d-9326-7e7b9115e26a.png">

Then in your CLI:
```bash
git clone https://github.com/londonsangongzi/obsidian-mindmap-nextgen/
```
When you're done:
```bash
# Stage all changes for commit
git add .
# create a commit in your local repository
git commit -m "fix: internal links break with &, | chars"
git push -u origin main
```
Then come back to this repo. Click <img width="153" alt="image" src="https://user-images.githubusercontent.com/10291002/219664012-dcbae9f3-17e9-40c6-9134-1b3c6fd2f000.png">

**Comment by londonsangongzi** [comment]
Hi James, thank you so much for your guidance on fixing this issue. I have submitted a pull request with the fix. Thanks again.

Jeff

**Comment by james-tindal** [comment]
You're welcome. Thanks for letting us know and contributing a fix.

**Comment by james-tindal** [comment]
Pushed your code in the latest version, @londonsangongzi

**Comment by londonsangongzi** [comment]
Hi, I noticed this issue is still present in the latest version 1.6.2, has anyone else encountered a similar problem with the latest release, thanks.

Jeff

**Comment by james-tindal** [comment]
Looks like I messed up the release!

**Comment by james-tindal** [comment]
Your code **is** included in version 1.6.3

**Comment by londonsangongzi** [comment]
Thanks, 1.6.3 works as expected.
